libs/types/src/tabulation.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  AnyContest,
+  BallotStyleId,
+  Candidate,
+  CandidateId,
+  ContestId,
+  PrecinctId,
+} from './election';
+import { Id } from './generic';
+
+export interface ContestResultsMetadata {
+  overvotes: number;
+  undervotes: number;
+  ballots: number;
+}
+
+type ContestResultsBase = ContestResultsMetadata & {
+  readonly contestId: ContestId;
+  readonly contestType: AnyContest['type'];
+};
+
+export type YesNoContestResults = ContestResultsBase & {
+  readonly contestType: 'yesno';
+  yesTally: number;
+  noTally: number;
+};
+
+export type CandidateTally = Candidate & {
+  tally: number;
+};
+
+export type CandidateContestResults = ContestResultsBase & {
+  readonly contestType: 'candidate';
+  readonly votesAllowed: number;
+  readonly tallies: Record<CandidateId, CandidateTally>;
+};
+
+/**
+ * Represents the results of a single contest in an election, often filtered by
+ * some cast vote record attributes.
+ */
+export type ContestResults = YesNoContestResults | CandidateContestResults;
+
+/**
+ * Should match `CVR.vxBallotStyle`.
```

Comment:
It's not possible to reference directly because the CVR type is auto-generated from the `vx-schema.json`, as an enum. I searched for ways to make a type out of the values of an enum but it seems that's not possible in Typescript Enums are just clunky in terms of extending or limiting them, which is why I didn't want to use one here.

Reply:
left this as is, but added a compile-time check in [8f67c3e](https://github.com/votingworks/vxsuite/pull/3519/commits/8f67c3ebbe6e2396d2e13aa3d6501684372a79af) and more detailed comment in [2fbda50](https://github.com/votingworks/vxsuite/pull/3519/commits/2fbda50cb98b367ac1b3fdab4d6c59609bf36331)